tests/Backend/SOX/MergeRequestsTest.php

```diff
@@ -155,6 +155,91 @@ public function testMRWorkflowFromDevelopmentToCancel(): void
         $this->assertNull($mrData['branches']['branchFromId']);
     }
 
+
+    public function testMRWorkflowLock(): void
+    {
+        $defaultBranch = $this->branches->getDefaultBranch();
+
+        // create two branches
+        $newBranch1 = $this->branches->createBranch($this->generateDescriptionForTestObject() . '_1');
+        $newBranch2 = $this->branches->createBranch($this->generateDescriptionForTestObject() . '_2');
+
+        // create same configuration in branches to create conflict
+        $configuration = (new Configuration())
+            ->setComponentId('wr-db')
+            ->setConfigurationId('main-1')
+            ->setName('Main')
+            ->setDescription('some desc');
+        (new Components(
+            $this->getBranchAwareClient($newBranch1['id'], [
+                'token' => STORAGE_API_DEVELOPER_TOKEN,
+                'url' => STORAGE_API_URL,
+            ])
+        ))->addConfiguration($configuration);
+        (new Components(
+            $this->getBranchAwareClient($newBranch2['id'], [
+                'token' => STORAGE_API_DEVELOPER_TOKEN,
+                'url' => STORAGE_API_URL,
+            ])
+        ))->addConfiguration($configuration);
+
+        // create MR for each branch
+        $mr1 = $this->developerClient->createMergeRequest([
+            'branchFromId' => $newBranch1['id'],
+            'branchIntoId' => $defaultBranch['id'],
+            'title' => 'Change everything',
+            'description' => 'Fix typo',
+        ]);
+        $mr2 = $this->developerClient->createMergeRequest([
+            'branchFromId' => $newBranch2['id'],
+            'branchIntoId' => $defaultBranch['id'],
+            'title' => 'Change everything',
+            'description' => 'Fix typo',
+        ]);
+
+        // set both MR to approved
+        $this->developerClient->mergeRequestPutToReview($mr1);
+        $this->developerClient->mergeRequestPutToReview($mr2);
+        $this->getReviewerStorageApiClient()->mergeRequestAddApproval($mr1);
+        $this->getReviewerStorageApiClient()->mergeRequestAddApproval($mr2);
+        $this->getSecondReviewerStorageApiClient()->mergeRequestAddApproval($mr1);
+        $this->getSecondReviewerStorageApiClient()->mergeRequestAddApproval($mr2);
+
+        // merge first MR
+        /** @var array{id: int} $mrJob */
+        $mrJob = $this->prodManagerClient->apiPutJson("merge-request/{$mr1}/merge", [], false);
```

Comment:
toto mi pride prave take nestabilne, ze spoliehas na to ze sa to este nestihne spravit. Ale ak to nahodou stihne prejst tak to na 229 spadne

Reply:
nechal bych to a uvidíme reálně jestli to bude padat, mohl bych tam k tomu dat komentář kdyby to spadlo.

Reply:
Trochu sem to okomentoval